gemfirexd/core/src/main/java/com/pivotal/gemfirexd/internal/engine/store/GemFireContainer.java

```diff
@@ -1931,7 +1931,7 @@ final void drop(final GemFireTransaction tran) throws StandardException {
           && (indexManager = (GfxdIndexManager)r.getIndexUpdater()) != null) {
         indexManager.drop(tran);
       }
-      if(r == null){
+      if(this.skipListMap != null){
         releaseIndexMemory();
       }
```

Comment:
So indexManager.drop itself does not do the memory accounting and is done explicitly by releaseIndexMemory?